timepiece/crm/models.py

```diff
@@ -1425,6 +1430,19 @@ def get_projects(self):
     def open_general_tasks(self):
         return self.generaltask_set.all() # status__terminal=False
 
+    def save(self, *args, **kwargs):
+        make_history = True
+        if self.pk is not None:
+            orig = Lead.objects.get(pk=self.pk)
+            if orig.status == self.status:
+                make_history = False
+
+        super(Lead, self).save(*args, **kwargs)
+
+        if make_history:
+            history = LeadHistory(lead=self,status=self.status)
```

Comment:
In the future just to `LeadHistory.objects.create(lead=self, status=self.status)`